### ISSUE TITLE
mnist-fashion results

### ISSUE THREAD

**Issue by maria8899** [adambielski/siamese-triplet#2]
Great work!

For mnist-fashion, I am surprised that it does not look like the contrastive/triplet are better than sotfmax, at least on the plot. I am even more surprised that the online sampling with negative mining does not seem to perform better than random sampling?
I understand you're using a 2d embedding space, but still. Do you have any quantitative numbers using maybe a higher dimensional space?
Could be interesting to plot in 3D as well.

**Comment by adambielski** [comment]
Hi @maria8899, thanks for your comment.
It all depends on how you define "better" :) For linear classification, of course the embeddings trained with softmax are better - that's what they're trained for. The plot might me somehow misleading, the range of features is much bigger (in order of hundreds) and you can't really see how many of the points actually overlap - actually the network trains to ~90% accuracy. I think we could get similar accuracy if we trained a 2 layer network on embeddings trained with triplet networks, but that of course should be validated.

As for online sampling with negative mining - I think the positive effect of this approach would be more visible in case of more complex datasets with many classes (e.g. face classification with many identities). But even for MNIST, the classes are better separated with this approach (compare the triplet loss for train set).

A good quantitative number would be - for example - to compare accuracy in a kNN setting on a test set. Didn't have time to do that yet, I might add this evaluation in the future.

3D plots would definitely be interesting, training a higher dimensional embeddings and visualizing them with t-sne would also be an interesting option.
